src/source-control/GitSourceControl.ts

```diff
@@ -263,64 +265,82 @@ export class GitSourceControl implements SourceControl {
         }
       });
-      queue.push(...baseCommitSttack.childCommits);
 
-      // // Update refs in our hashMap
+      // Update refs in our hashMap
       this.repo = produce(this.repo, (draftRepo) => {
-        const draftTargetCommitSttack = draftRepo.commits.get(targetCommit!)!;
-        const childrenOfTargetCommit = draftTargetCommitSttack.childCommits;
+        const draftTargetCommit = draftRepo.commits.get(targetCommitHash)!;
+        const childrenOfTargetCommit = draftTargetCommit.childCommits;
+
+        // Add the newly-created commit to our repo
         const newCommitToInsert: Commit = {
           hash: newCommitOid.tostrS(),
-          title: baseCommitSttack.title,
+          title: commitToBeRebased.title,
           timestamp: new Date(),
-          author: baseCommitSttack.author,
-          committer: baseCommitSttack.committer,
-          refNames: baseCommitSttack.refNames.filter(refIsLocal),
-          parentCommits: [draftTargetCommitSttack.hash],
+          author: commitToBeRebased.author,
+          committer: commitToBeRebased.committer,
+          refNames: commitToBeRebased.refNames.filter(refIsLocal),
+          parentCommits: [draftTargetCommit.hash],
           childCommits: [...childrenOfTargetCommit],
         };
         draftRepo.commits.set(newCommitOid.tostrS(), newCommitToInsert);
 
-        // Update ChildCommitHash for targetCommitSttack to newCommit
-        draftTargetCommitSttack.childCommits = [newCommitOid.tostrS()];
-        // Update ParentCommitHash for all of the children and replace targetCommitSttack's hash with newCommit's hash
-        childrenOfTargetCommit.forEach((child) => {
-          const childCommit = draftRepo.commits.get(child)!;
-          const parentsOfChild = childCommit.parentCommits.map((parent) => {
-            if (parent === draftTargetCommitSttack.hash) {
-              return newCommitOid.tostrS();
-            }
-            return parent;
-          });
-          childCommit.parentCommits = parentsOfChild;
-          draftRepo.commits.set(child, childCommit);
+        // Update target's children
+        draftTargetCommit.childCommits = [newCommitOid.tostrS()];
+
+        // Update ParentCommitHash for all of the children and replace targetCommit's hash with newCommit's hash
+        childrenOfTargetCommit.forEach((childHash) => {
+          const childCommit = draftRepo.commits.get(childHash)!;
+          const newParentCommitHashes = childCommit.parentCommits.map(
+            (parent) => {
+              // Replace pre-cherry-pick parent with the post-cherry-pick one
+              if (parent === draftTargetCommit.hash) {
+                return newCommitOid.tostrS();
+              }
+              return parent;
+            },
+          );
+          childCommit.parentCommits = newParentCommitHashes;
+          draftRepo.commits.set(childHash, childCommit);
         });
       });
+
+      queue.push(...commitToBeRebased.childCommits);
```

Comment:
Is there a reason why this was moved to after all the adjusting on commit hashes? I did this before in line `285`, basically before all the ref names altering. I think this does the same thing right?

Reply:
Yep, there shouldn't be any observable change in behaviour. I just moved it down here so that all the code that prepares for the next iteration is in one place. I thought that'll be cleaner, but we can move it back if you disagree

Reply:
Nah, that's okay, I was just trying to identify if there is any change because of this